src/Pdf.php

```diff
@@ -12,38 +13,64 @@ class Pdf
 
     protected $binPath;
 
+    protected $options = [];
+
     public function __construct(string $binPath = null)
     {
         $this->binPath = $binPath ?? '/usr/bin/pdftotext';
     }
 
-    public function setPdf(string $pdf) : Pdf
+    public function setPdf(string $pdf) : self
     {
-        if (!file_exists($pdf)) {
-            throw new PdfNotFound("could not find pdf {$pdf}");
+        if (!\is_readable($pdf)) {
+            throw new PdfNotFound(\sprintf('could not find or read pdf `%s`', $pdf));
         }
 
         $this->pdf = $pdf;
 
         return $this;
     }
 
+    public function setOptions(array $options) : self
+    {
+        $this->options = \array_map([$this, 'filterOption'], $options);
```

Comment:
The filtering of the options is not needed. Let's make it so that the users doesn't have to prefix options with `-`

Reply:
I added the option to add or not the starting hyphen .. if present it won't be added if not it will be added. the bonus is that the array is self validated by PHP so if you an array which contains something which is not stringable it will throw a `TypeError`.